client/apps/edit/components/section_image_set/index.coffee

```diff
@@ -0,0 +1,263 @@
+#
+# Image Set section that allows uploading a mix of images and artworks
+#
+
+# Using `try` here b/c Scribe is an AMD module that doesn't play nice when
+# requiring it for testing in node.
+try
+  Scribe = require 'scribe-editor'
+  scribePluginToolbar = require 'scribe-plugin-toolbar'
+  scribePluginSanitizer = require '../../lib/sanitizer.coffee'
+  scribePluginLinkTooltip = require 'scribe-plugin-enhanced-link-tooltip'
+_ = require 'underscore'
+gemup = require 'gemup'
+React = require 'react'
+toggleScribePlaceholder = require '../../lib/toggle_scribe_placeholder.coffee'
+sd = require('sharify').data
+icons = -> require('./icons.jade') arguments...
+Autocomplete = require '../../../../components/autocomplete/index.coffee'
+Artwork = require '../../../../models/artwork.coffee'
+{ div, section, h1, h2, span, img, header, input, nav, a, button, p, ul, li, strong } = React.DOM
+{ crop, resize, fill } = embedly = require('embedly-view-helpers')(sd.EMBEDLY_KEY)
+
+module.exports = React.createClass
+
+  getInitialState: ->
+    images: @props.section.get('images') or []
+    progress: null
+
+  componentDidMount: ->
+    @attachScribe()
+    @setupAutocomplete()
+    @showPreviewImages()
+
+  componentDidUpdate: ->
+    @attachScribe()
+
+  componentWillUnmount: ->
+    @autocomplete.remove()
+
+  onStateChange: ->
+    @props.section.set images: @state.images if @state.images.length > 0
+    @showPreviewImages()
+    @forceUpdate()
```

Comment:
You also should have to use `forceUpdate` unless you're hooking into something un-react-like (e.g. events on Backbone models). I bet if you refactor the state mutations into `setState` you'll notice you won't need this.

Also if `onStateChange` is meant to run after every "re-render" than I think a [lifecycle callback](https://facebook.github.io/react/docs/component-specs.html#updating-componentdidupdate) would be more appropriate.